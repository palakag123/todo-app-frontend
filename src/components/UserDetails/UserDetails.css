.details-container{
    width: 35%;
    background-color: white;
    height: 100vh;
    padding:5px;
    display: flex;
    justify-content: center;
    align-items: center;
    overflow-y: hidden;
    overflow-x: hidden;
}
.details-inner-container{
    border: 3px solid rgb(143, 237, 244);
    height: 97vh;
    width: 95%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    overflow-y: hidden;
    overflow-x: hidden;
}
#user-img{
    border-radius: 50%;
}
#greeting{
    color: rgb(54, 195, 240);
}

@media only screen and (max-width: 770px) {
    .details-container{
        width: 100vw;
        background-color: white;
        height: 50vh;
        padding: 0px;
    }
    .details-inner-container{
        height: 90%;
        width: 90%;
        margin: 0px;
    }
    #user-img{
        width: 170px;
    }
    #greeting{
        font-size: 20px;
    }
    #date-time{
        font-size: 20px;
    }
}
@media only screen and (max-width: 560px) {
    
    #user-img{
        width: 100px;
    }
    
}