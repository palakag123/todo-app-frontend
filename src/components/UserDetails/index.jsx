/* eslint-disable no-nested-ternary */
/* eslint-disable operator-linebreak */
/* eslint-disable comma-dangle */
import React from "react";
import "./UserDetails.css";
import Clock from "react-live-clock";
import userImage from "../../resources/userImage.jpeg";

function UserDetails() {
  return (
    <div className="details-container">
      <div className="details-inner-container">
        <img id="user-img" src={userImage} alt="user" data-testid="userImg" />
        <h2 id="greeting" data-testid="greeting">
          HI, USER
        </h2>
        <br />
        <h1 id="date-time" data-testid="time">
          <Clock format="h:mm a" ticking timezone="Asia/Calcutta" />
        </h1>
        <h1 id="date-time" data-testid="date">
          <Clock ticking format="DD MMMM, YYYY" timezone="Asia/Calcutta" />
        </h1>
      </div>
    </div>
  );
}

export default UserDetails;
