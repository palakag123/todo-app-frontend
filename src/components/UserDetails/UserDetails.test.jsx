import React from "react";
import { render, screen } from "@testing-library/react";
import UserDetails from "./index";

describe("UserDetails", () => {
  beforeEach(() => {
    render(<UserDetails />);
  });
  it("should take a snapshot for UserDetails component", () => {
    const { asFragment } = render(<UserDetails />);
    expect(asFragment(<UserDetails />)).toMatchSnapshot();
  });
  it("should have elements required when rendered", () => {
    expect(screen.getByTestId("userImg")).toBeTruthy();
    expect(screen.getByTestId("greeting")).toBeTruthy();
    expect(screen.getByText("HI, USER")).toBeTruthy();
    expect(screen.getByTestId("time")).toBeTruthy();
    expect(screen.getByTestId("date")).toBeTruthy();
  });
});
