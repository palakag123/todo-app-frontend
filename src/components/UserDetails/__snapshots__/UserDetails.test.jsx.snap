// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails should take a snapshot for UserDetails component 1`] = `
<DocumentFragment>
  <div
    class="details-container"
  >
    <div
      class="details-inner-container"
    >
      <img
        alt="user"
        data-testid="userImg"
        id="user-img"
        src="userImage.jpeg"
      />
      <h2
        data-testid="greeting"
        id="greeting"
      >
        HI, USER
      </h2>
      <br />
      <h1
        data-testid="time"
        id="date-time"
      >
        <time
          datetime="1650831384718"
        >
          1:46 am
        </time>
      </h1>
      <h1
        data-testid="date"
        id="date-time"
      >
        <time
          datetime="1650831384718"
        >
          25 April, 2022
        </time>
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
